### ISSUE TITLE
Google Colab NOTEBOOK FOR TRAINING AND PREDICTION

### ISSUE THREAD

**Issue by ekesdf** [alic-xc/issues#2]
[Noteboook](https://colab.research.google.com/drive/1e7tVMVEYFln2WqT6N7-vdFdh_to-Pd6H?usp=sharing)

@alic-xc 
you can edit so just fill in the space holder and run the **Downloads and GPU check** Cell and then the other Cell one at a time depending on what you want to do, **but the first Cell must always be executed**

**Comment by alic-xc** [comment]
Okay. Good

**Comment by alic-xc** [comment]
Check this out.

![ee](https://user-images.githubusercontent.com/23708960/122800314-b194c880-d2ba-11eb-8789-0757d1e6cffe.jpeg)

For the accuracy, it is detecting 2 fishes as 1, how can i solve that ?

![test2](https://user-images.githubusercontent.com/23708960/122800323-b3f72280-d2ba-11eb-9baf-e7851d8aa831.jpg)

**Comment by ekesdf** [comment]
And now to your Questions

1. You can improve that by training the model with images where there are multiple fishes in the same frame.

2. The bad detections results are due to the "bad" training all in all it went well, but the model is just trained to segment one fish and not multiple fishes at once so like I said already at 1. Just retrain the model and it should work:)
3. and 4. You can retrain and improve the model on your own using the Notebook
5. I can't give a precise prediction of how high the end ACC would be, but if you train with "real life" images (images you want the model to detect in the real world) it should get at least 80 if the amount of training data is enough

**Comment by ekesdf** [comment]
> Check this out.
> 
> ![ee](https://user-images.githubusercontent.com/23708960/122800314-b194c880-d2ba-11eb-8789-0757d1e6cffe.jpeg)
> 
> For the accuracy, it is detecting 2 fishes as 1, how can i solve that ?
> 
> ![test2](https://user-images.githubusercontent.com/23708960/122800323-b3f72280-d2ba-11eb-9baf-e7851d8aa831.jpg)

this should be fixable by a more accurate labeling of those fishes

**Comment by ekesdf** [comment]
How many images you trained with ??

**Comment by alic-xc** [comment]
300 images

**Comment by ekesdf** [comment]
So the dataset you sent me ?

**Comment by alic-xc** [comment]
yes

**Comment by alic-xc** [comment]
So, to get a better accuracy, i just need to get images with multiple fishes right ?

**Comment by ekesdf** [comment]
yea

**Comment by ekesdf** [comment]
Bc the trained model performs only well on the data it is trained

**Comment by alic-xc** [comment]
Oh.

**Comment by alic-xc** [comment]
> Bc the trained model performs only well on the data it is trained

so, to be able to capture all possible scenario, i just need to train more images.

**Comment by ekesdf** [comment]
Yea and more images with multiple fishes bc in the end there will be probably more images with more than lets say 2 fishes 
or did I get this wrong ?

**Comment by alic-xc** [comment]
i don't understand

**Comment by ekesdf** [comment]
You will make prediction on images with more than 2 fishes right

**Comment by alic-xc** [comment]
![earlyFingerling372](https://user-images.githubusercontent.com/23708960/122803633-c1aea700-d2be-11eb-9cb4-0aa330ba87f2.jpg)

This for example right ?

**Comment by ekesdf** [comment]
right

**Comment by alic-xc** [comment]
> You will most likely make prediction on images with more than 2 fishes, right?

Yes.

**Comment by ekesdf** [comment]
so it would make sense to train on images like that ?

**Comment by alic-xc** [comment]
Yes, i understand now.
So i should train more of images like that.

**Comment by ekesdf** [comment]
yea :) :partying_face:

**Comment by alic-xc** [comment]
Great.

**Comment by alic-xc** [comment]
Thanks.
It's a great pleasure meeting you.

**Comment by ekesdf** [comment]
The pleasure is on my site I love to share my knowledge

**Comment by ekesdf** [comment]
3 Ideas

I can help you labeling these Images 

I can help you train the model bc I have my tesla k80 at home so I have 2 11GB GPUs to train at

Do you need my help any help or do you make it from here on solo 

@alic-xc

**Comment by alic-xc** [comment]
I would really appreciate.

Let me brief the process, so you can decide any help you can offer.
The project is based on fingerlings(Fish) counting system which i want to be like 98% accurate which i doubt.

The dataset which i have sent to you are the files i have labelled but you can crosscheck it if they are labeled properly, but most images contain single fish. So, i have over 10-30 images that contain multiple fishes. Would you mind if i could send them ? 

Also, if you have any suggestion about this projects, feel free to discuss with me.

At the end of everything, i just want a model that have a better accuracy.

**Comment by SB2020-eye** [comment]
I saw the first thread over at [ImageAI](https://github.com/OlafenwaMoses/ImageAI/issues/665), and then sought out the rest of how things have progressed here.

I'm impressed by the progress.

I have a project and I think your method may help me. (I am very new to coding, and really it's towards the goal of something more like an art project.)

Just to make sure: you both used PixelLib, correct?

Also, were the steps pretty straightforward?

Will I be able to do this without GPU?

My images look like this:
![Chad222-m](https://user-images.githubusercontent.com/56878280/122853496-4eb62800-d2e0-11eb-8dac-e1efbe5a2a63.jpg)

I need to cut out all the letters -- for example, this:
![Chad_one_word e](https://user-images.githubusercontent.com/56878280/122853565-6d1c2380-d2e0-11eb-83c4-91d821d10793.png)

Suggestions welcome!

**Comment by ekesdf** [comment]
Hi @SB2020-eye 

I know a way we can make this work. I like the idea of alic-xc to create a separate repo to discuss the project. So it would be nice if you could do that to so we separate those two projects

**Comment by ekesdf** [comment]
> I would eat it.
> 
> Let me brief the process, so you can decide on any help you can offer.
> The project is based on fingerlings(Fish) counting system which I want to be like 98% accurate which I doubt.
> 
> The dataset which I have sent to you are the files I have labeled but you can crosscheck it if they are labeled properly, but most images contain single fish. So, I have over 10-30 images that contain multiple fishes. Would you mind if I could send them? 
> 
> Also, if you have any suggestion about this projects, feel free to discuss with me.
> 
> At the end of everything, I just want a model that has better accuracy.

I have no problem with you send them to me but is it possible to upload them. To GitHub so we can share files easier

**Comment by ekesdf** [comment]
@alic-xc 

I have no problem looking at your labeled dataset and checking your labels

It would be easier to upload them to GitHub rather than sending me them in an other way

**Comment by alic-xc** [comment]
I think it will be better if I upload it as a zip file

**Comment by ekesdf** [comment]
No with that

**Comment by alic-xc** [comment]
This is the dataset, 
[fingerlings.zip](https://github.com/alic-xc/issues/files/6693651/fingerlings.zip)

This is the new images with multiple fishes
[testing.zip](https://github.com/alic-xc/issues/files/6693592/testing.zip)

**Comment by SB2020-eye** [comment]
> Hi @SB2020-eye
> 
> I know a way we can make this work. I like the idea of alic-xc to create a separate repo to discuss the project. So it would be nice if you could do that to so we separate those two projects

Wonderful! [Done](https://github.com/SB2020-eye/Issues/issues/1).

**Comment by alic-xc** [comment]
@ekesdf 
i just check my google repo and i found some images for the fish(fingerling), but i don't know if they will be useful do you mind me sending it ?

**Comment by ekesdf** [comment]
Hi
@alic-xc what do you think about the idea of uploading all the images you want me to review into this GitHub repo 
For example, one folder unlabeled one folder labeled. And one where you are not sure about the usefulness of the image

**Comment by ekesdf** [comment]
> @ekesdf 
> I just check my google drive and I found some images for the fish(fingerling), but I don't know if they will be useful do you mind me sending them?
> 

And yeah I would be fine with it

**Comment by alic-xc** [comment]
> Hi
> @alic-xc what do you think about the idea of uploading all the images you want me to review into this GitHub repo
> For example, one folder unlabeled one folder labeled. And one where you are not sure about the usefulness of the image

@ekesdf 
What i was thinking was that, you can help review the dataset(first folder) i have labeled to see if it is well label. 
So, about building model that can help predicting a good results on multiple fishes in an image, that was why i  uploaded the unlabeled dataset (second folder) folder one.
So, the images i found in my google drive are also fish(fingerlings) images, i just thought of sharing it, so that it can help when training the model.

Or is the workload too much ?
which one do you think you can achieve ?
what part do you think i can play ?

**Comment by ekesdf** [comment]
Well you can do everything on you're own so you can label and check them for that I can give you an example of what is right and what is wrong.
For the effort labeling, I can help you with that

**Comment by alic-xc** [comment]
Okay, that's great

**Comment by ekesdf** [comment]
Hey, how it is going ? 
I know it has been some time we wrote last. I had a bit of stress bc I had to finish my A-levels, so now I have time working on some projects after a graduated successfully. So I was wondering if you still work on your project, or do you discarded it ?

**Comment by alic-xc** [comment]
Oh, great, congratulation.
Though, i have not abandon the project. 
would you mind us chatting through whatsapp or slack?
By doing that, it will be faster and easier to communicate. i have ton of questions to ask

**Comment by ekesdf** [comment]
np talking over whatsapp
+4915236644716

**Comment by alic-xc** [comment]
Great, i will chat you up in a moment.
Thanks so much

**Comment by ekesdf** [comment]
So I can delete my number ??

**Comment by alic-xc** [comment]
i have deleted it

**Comment by ekesdf** [comment]
thx

**Comment by ekesdf** [comment]
```
from imageai.Detection.Custom import DetectionModelTrainer

trainer = DetectionModelTrainer()
trainer.setModelTypeAsYOLOv3()
trainer.setDataDirectory(data_directory="hololens")
metrics = trainer.evaluateModel(model_path="hololens/models", json_path="hololens/json/detection_config.json", iou_threshold=0.5, object_threshold=0.3, nms_threshold=0.5)
print(metrics)
```
